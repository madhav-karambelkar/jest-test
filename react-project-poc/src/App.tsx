import './App.css'
import { useAppDispatch, useAppSelector } from './store/hook'
import { selectCurrentCount, setDecreaseCount, setIncreaseCount } from './store/features/count/countSlice'

function App() {
  const currentCount = useAppSelector(selectCurrentCount)
  const dispatch = useAppDispatch()

  return (
    <>
      <h1>Vite + React</h1>
      <div className="card">
        <p data-testid="current_count">count is {currentCount}</p>
        <button onClick={() => {
          dispatch(setIncreaseCount(1))
        }}>
          Add
        </button>
        <button onClick={() => {
          dispatch(setDecreaseCount(1))
        }}>
          Subtract
        </button>
        <p>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
      </div>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </>
  )
}

export default App
