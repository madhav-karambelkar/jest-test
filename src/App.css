body{
  background: linear-gradient(45deg, rgb(133, 255, 189) 31%, rgb(255, 251, 125) 100%);
  background-size: cover;
  height: 100vh;
}