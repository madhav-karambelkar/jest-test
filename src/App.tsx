import { useEffect, useState } from "react";
import "./App.css";

function App() {


const [user, setUser] = useState([])

 useEffect(()=>{
  getUserData()
 }, [])

 const getUserData = async () => {
  const result = await fetch('https://randomuser.me/api/')
  const res = await result.json()
  console.log(res?.results)
  setUser(res.results)
 }
 
  return (
    <div className="App mt-5">
      <div className="container">
        {user?.length && user.map((userData:any, idx:any)=>(
           <div className="row" key={userData.cell}>
           <div className="col-12">
             <img src={userData.picture.large} alt="" className="img-fluid" height={200}/>
           </div>
           <div className="col-12">
             <div className=" mt-2">
               <div className="h3 text-primary mr-4">
                 Cell :
               </div>
               <div className="h4 text-secondary">
                 {userData.cell}
               </div>
             </div>
           </div>
           <div className="col-12">
             <div className="h3 text-primary mr-4">
               <div>
                 DOB :
               </div>
               <div className="h4 text-secondary">
                 {userData.dob.date}
               </div>
             </div>
           </div>
           <div className="col-12">
             <div className="h3 text-primary mr-4">
               <div>
                 Email :
               </div>
               <div className="h4 text-secondary">
                 {userData.email}
               </div>
             </div>
           </div>
           <div className="col-12">
             <div className="h3 text-primary mr-4">
               <div>
                 Gender :
               </div>
               <div className="h4 text-secondary">
                 {userData.gender}
               </div>
             </div>
           </div>
         </div>
        ))}
       
      </div>
    </div>
  );
}

export default App;
