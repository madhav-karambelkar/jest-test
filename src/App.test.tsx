import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';

const consoleSpy = jest.spyOn(console, "log")

describe('Renders App Component', ()=>{
  afterEach(()=>{
    jest.clearAllMocks()
  })

  test('it should render sign up card with text Sign Up Form', () => {
    render(<App />);
    const titleText = screen.getByText('Sign Up Form')
    expect(screen).toBeTruthy();
    expect(titleText).toBeInTheDocument()
  });

  test('it should render email input inside sign up card with its properties', () => {
    render(<App />);

    const inp = screen.getByLabelText(/email address/i)
    
    expect(inp).toBeInTheDocument()
    expect(inp).toHaveAttribute('name', 'email')
    expect(inp).toHaveAttribute('id', 'email')
    expect(inp).toHaveAttribute('type', 'text')
    expect(inp).toHaveAttribute('placeholder', 'Email Address')

  });

  test('it should render fullname input inside sign up card with its properties', () => {
    render(<App />);

    const inp = screen.getByLabelText('Name')

    expect(inp).toBeInTheDocument()
    expect(inp).toHaveAttribute('name', 'name')
    expect(inp).toHaveAttribute('id', 'name')
    expect(inp).toHaveAttribute('type', 'text')
    expect(inp).toHaveAttribute('placeholder', 'Full Name')
    
  });

  test('it should render password input inside sign up card with its properties', () => {
    render(<App />);

    const inp = screen.getByLabelText('Password')

    expect(inp).toBeInTheDocument()
    expect(inp).toHaveAttribute('name', 'password')
    expect(inp).toHaveAttribute('id', 'password')
    expect(inp).toHaveAttribute('type', 'password')
    expect(inp).toHaveAttribute('placeholder', 'Password')
    
  });

  test('it should render city input inside sign up card with its properties', () => {
    render(<App />);

    const inp = screen.getByLabelText('City')

    expect(inp).toBeInTheDocument()
    expect(inp).toHaveAttribute('name', 'city')
    expect(inp).toHaveAttribute('id', 'city')
    expect(inp).toHaveAttribute('type', 'text')
    expect(inp).toHaveAttribute('placeholder', 'City')

  });

  test('it should render submit button inside sign up card with its properties', () => {
    render(<App />);

    const btn = screen.getByRole('button')

    expect(btn).toBeInTheDocument()
    expect(btn).toHaveAttribute('class', 'btn btn-primary w-100 mt-5')
    expect(btn).toHaveAttribute('type', 'submit')
  });

  test('it should submit the form while clicking on submit button', () => {

    render(<App />);
    const btn = screen.getByRole('button', {name: 'Submit'})

    expect(btn).toBeInTheDocument()

    fireEvent.click(btn)

    expect(consoleSpy).toReturn()
    expect(consoleSpy).toBeCalledWith('Form Submitted')
    
  });

  test('it should check on change method', () => {

    render(<App />);
    const inp: any = screen.getByLabelText('City')
    fireEvent.change(inp, {target: {value: 'Indore'}})
    // userEvent.type(inp, 'Indore')


    expect(inp.value).toBe('Indore')
    
  });

  test('it should match snapshot', () => {

   const {container} = render(<App />);
   
   expect(container).toMatchSnapshot()
    
 });
})
