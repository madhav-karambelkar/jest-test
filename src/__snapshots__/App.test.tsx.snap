// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App Component it should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-12 col-sm-10 col-md-8 col-lg-6 mx-auto"
        >
          <div
            class="card w-100 my-5 p-5"
          >
            <h1>
              Sign Up Form
            </h1>
            <form
              action=""
              class="mt-3"
            >
              <label
                class="mt-3 d-flex ml-1 w-100"
                for="name"
              >
                Name
              </label>
              <input
                class="form-control"
                id="name"
                name="name"
                placeholder="Full Name"
                type="text"
                value=""
              />
              <label
                class="mt-3 d-flex ml-1 w-100"
                for="email"
              >
                Email Address
              </label>
              <input
                class="form-control"
                id="email"
                name="email"
                placeholder="Email Address"
                type="text"
                value=""
              />
              <label
                class="mt-3 d-flex ml-1 w-100"
                for="password"
              >
                Password
              </label>
              <input
                class="form-control"
                id="password"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
              <label
                class="mt-3 d-flex ml-1 w-100"
                for="city"
              >
                City
              </label>
              <input
                class="form-control"
                disabled=""
                id="city"
                name="city"
                placeholder="City"
                type="text"
                value=""
              />
              <button
                class="btn btn-primary w-100 mt-5"
                type="submit"
              >
                 Submit 
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
